pkg/gateway/model_build_targets.go

```diff
@@ -107,57 +108,75 @@ func (t *latticeTargetsModelBuildTask) buildLatticeTargets(ctx context.Context)
+		err = t.Client.Get(ctx, namespacedName, serviceExport)
 		if err != nil {
-			glog.V(6).Infof("Failed to read Annotaions/Port:%v, err:%s\n ", serviceExport.ObjectMeta.Annotations[portAnnotationsKey], err)
+			glog.V(6).Infof("Failed to find Service export in the DS. Name:%s, Namespace:%s - err:%s", t.tgName, t.tgNamespace, err)
+		} else {
+			portsAnnotations := strings.Split(serviceExport.ObjectMeta.Annotations[portAnnotationsKey], ",")
+
+			for _, portAnnotation := range portsAnnotations {
+				definedPort, err := strconv.ParseInt(portAnnotation, 10, 64)
+				if err != nil {
+					glog.V(6).Infof("Failed to read Annotations/Port:%s, err:%s", serviceExport.ObjectMeta.Annotations[portAnnotationsKey], err)
+				} else {
+					definedPorts = append(definedPorts, definedPort)
+				}
+			}
+			glog.V(6).Infof("Build Targets - portAnnotations: %v", definedPorts)
 		}
-		glog.V(6).Infof("Build Targets - portAnnotations: %v \n", portAnnotations)
+	} else if tg.ByBackendRef {
+		definedPorts = getDefinedPortsFromK8sRouteRules(t, definedPorts)
+	}
+	if len(definedPorts) == 0 {
+		definedPorts = []int64{undefinedPort}
 	}
 
 	var targetList []latticemodel.Target
 	endPoints := &corev1.Endpoints{}
 
 	if svc.DeletionTimestamp.IsZero() {
 		if err := t.Client.Get(ctx, namespacedName, endPoints); err != nil {
-			errmsg := fmt.Sprintf("Build Targets failed because K8S service %v does not exist", namespacedName)
-			glog.V(6).Infof("errmsg: %v\n", errmsg)
+			errmsg := fmt.Sprintf("Build Targets failed because K8S service %s does not exist", namespacedName)
+			glog.V(6).Infof("errmsg: %s", errmsg)
 			return errors.New(errmsg)
 		}
 
-		glog.V(6).Infof("Build Targets:  endPoints %v \n", endPoints)
+		glog.V(6).Infof("Build Targets:  endPoints %s", endPoints)
 
 		for _, endPoint := range endPoints.Subsets {
 			for _, address := range endPoint.Addresses {
 				for _, port := range endPoint.Ports {
-					glog.V(6).Infof("serviceReconcile-endpoints: address %v, port %v\n", address, port)
+					glog.V(6).Infof("serviceReconcile-endpoints: address %v, port %v", address, port)
 					target := latticemodel.Target{
 						TargetIP: address.IP,
 						Port:     int64(port.Port),
 					}
-					if portAnnotations == undefinedPort || int64(target.Port) == portAnnotations {
-						targetList = append(targetList, target)
-						glog.V(6).Infof("portAnnotations:%v, target.Port:%v\n", portAnnotations, target.Port)
-					} else {
-						glog.V(6).Infof("Found a port match, registering the target - port:%v, containerPort:%v, taerget:%v ***\n", int64(target.Port), portAnnotations, target)
+
+					for _, definedPort := range definedPorts {
+						if target.Port == definedPort {
+							// target port matches service export port
+							targetList = append(targetList, target)
+							glog.V(6).Infof("portAnnotations:%d, target.Port:%d", definedPort, target.Port)
+						} else if definedPort == undefinedPort || definedPort == target.Port {
```

Comment:
I still don't get this condition, seems `definedPort == target.Port` is covered in the above condition already?

nit: the debug logs here seem a bit overkill

nit: on `// target port matches service export port` - it could be httproute not service export